feat(linter): add suggestions for `no-plusplus`

camchenry opened the pull request:
Implements naive suggestions for the `no-plusplus` rule. This is not implemented as a fix, because there isn't an exact 1:1 equivalence between `++` and `+= 1`. But most cases should be autofixable by this, but I've marked it as dangerous suggestion due to the semantics change

camchenry commented:
* **#6376** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6376?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @camchenry and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

DonIsaac commented on crates/oxc_linter/src/rules/eslint/no_plusplus.rs:
```diff
@@ -78,7 +78,10 @@ declare_oxc_lint!(
+    // This is "dangerous" because this will rewrite the code in a way that is not equivalent.
+    // For example, `++i` and `i++` will be rewritten as `i += 1` even though they are not the same.
+    // If the code depends on the order of evaluation, then this might break it.
+    dangerous_suggestion,
```
```suggestion
    conditional_dangerous_suggestion,
```

DonIsaac replied:
We could even make this `conditional_suggestion`, since this is unlikely to create semantically/syntactically invalid JavaScript.

camchenry replied:
fair, there is part of me that fears this creating some really strange regressions, but I guess labeling it as a "suggestion" already implies it's not a guaranteed fix

camchenry committed:
feat(linter): add suggestions for `no-plusplus` (#6376)

Implements naive suggestions for the `no-plusplus` rule. This is not implemented as a fix, because there isn't an exact 1:1 equivalence between `++` and `+= 1`. But most cases should be autofixable by this, but I've marked it as dangerous suggestion due to the semantics change
